### ISSUE TITLE
[BUG] wasmcloud logs don't report MemoryOutOfBounds error

### ISSUE THREAD

**Issue by cdmurph32** [wasmCloud/wasmCloud#3050]
### Affected project(s)

- [ ] documentation
- [ ] examples
- [X] wasmCloud host
- [ ] wasmCloud CLI (wash)
- [ ] wasmCloud dashboard UI (washboard)
- [ ] capability providers
- [ ] provider bindgen
- [ ] control interface client
- [ ] other / not sure

### Describe the bug

Nothing is logged If a component hits a MemoryOutOfBounds error.

### Steps to reproduce

1. Remove `-z stack-size=2097152` from https://github.com/cdmurph32/pdf-text-example/blob/main/Makefile#L15
2. make and run pdf-text-example
3. Remove call to `print_pdf_text` https://github.com/cdmurph32/pdf-text-example/blob/main/pdf_text.c#L230, this doesn't work for different reasons.
4. Post a PDF to the service `curl -X POST --data-binary "@pdfio/testfiles/testpdfio.pdf" localhost:8080` to hit a MemoryOutOfBounds error

### Expected behavior

I expect the an error to be in the wasmcloud logs.

### Environment

- OS: MacOS
- Shell zsh
- wasmcloud Version v1.2.1

**Comment by rvolosatovs** [comment]
We just found out that component export handling failures were never logged and fixed that in https://github.com/wasmCloud/wasmCloud/pull/3060

@cdmurph32 could you test if there's now an error reported if you use wasmCloud from latesr `main`?

**Comment by cdmurph32** [comment]
There is now this message in the logs.
```
WARN wasmcloud_host::wasmbus: failed to serve invocation err=failed to serve `wrpc:http/incoming-handler@0.1.0.handle` invocation
```
commit `ad8aecddaf097ae678518e5b3be681e1cb8b5132`

**Comment by cdmurph32** [comment]
It would be nice if the log could give the same output as wasmtime. These types of errors could be completely dependent on the payload, in this case the size of the payload. It would help debugging production issues.

**Comment by rvolosatovs** [comment]
Since https://github.com/wasmCloud/wasmCloud/pull/3117 is merged, we should be logging the Wasmtime error as well, could you confirm that all the output you expect is now available in latest `main`?
You should now see up to two log statements on errors here

**Comment by cdmurph32** [comment]
Looks like this was fixed with commit: `f5093226`

```
2024-10-02T14:11:14.570306Z  WARN wasmcloud_runtime::component::http: failed to call `wasi:http/incoming-handler.handle` err=error while executing at wasm backtrace:
    0: 0xb07d - .tmpWA4neV!load_xref
    1: 0xaf7e - .tmpWA4neV!pdfioMemBufOpen
    2: 0x5ce5 - .tmpWA4neV!show_pdf_info
    3: 0x6b2c - .tmpWA4neV!exports_wasi_http_incoming_handler_handle
    4: 0x5c67 - .tmpWA4neV!__wasm_export_exports_wasi_http_incoming_handler_handle
note: using the `WASMTIME_BACKTRACE_DETAILS=1` environment variable may show more debugging information

Caused by:
    0: memory fault at wasm address 0x100000a60 in linear memory of size 0xd0000
    1: wasm trap: out of bounds memory access
```
